src/torchcodec/decoders/_core/VideoDecoder.cpp

```diff
@@ -856,6 +856,25 @@ VideoDecoder::DecodedOutput VideoDecoder::convertAVFrameToDecodedOutput(
   output.duration = getDuration(frame);
   output.durationSeconds = ptsToSeconds(
       getDuration(frame), formatContext_->streams[streamIndex]->time_base);
+  if (streamInfo.options.device.type() != torch::kCPU) {
```

Comment:
Is this logic truly general to all non-CPU devices? If no, then here and elsewhere, we should do something closer to:
```
if (streamInfo.options.device.type() == torch::kCUDA) {
  logicSpecificToCUDA();
}
else if (streamInfo.options.device.type() == torch::kCPU) {
  logicSpecificToCPU();
}
else {
  TORCH_CHECK(false, "Unsupported device");
}
```

Reply:
Right now we only support cuda.

I am assuming if we support AMD we will use the same interface, and use cmake or #ifdefs to link in the correct device code.

So VideoDecoder.cpp just assumes cmake or the linker will do the right thing and just calls the device code for any type of device.

At the moment cuda device is linked by cmake for cuda builds. How that will be done for AMD is TBD.

Reply:
I see, that makes sense. That also means that we don't need to have CPU versions tof functions hat throw for all N devices we support.